content/en/blog/2024-07-25-argocd-and-flux-vs-kluctl.md

```diff
@@ -0,0 +1,469 @@
+implement more complex deployments, you usually have to use multiple Custom Resources together. Sometimes you have to even chain them (e.g.
+Kustomization -> Kustomization -> HelmRelease). Please note that each Kustomization (the CRs) in this chain also might
+imply a `kustomization.yaml` that can also use overlays and components which might lead to even longer chains of CRs. A
+HelmRelease could potentially also deploy Kustomizations or other HelmReleases, when dynamic configuration via templating
+is desired.
+
+Following and fully understaning these chains can become quite challenging, when the project size and complexity grows.
+
+Another implication of using CRs is that your deployments become 100% dependent on the controllers running
+inside Kubernetes, because in-cluster reconciliation is the only way to process the CRs. This means, you can not fully
+test or verify your deployments before pushing them to your Git repository. The only way to reliably prevent killing your production
+environment is to introduce testing/staging environments, adding even more complexity to your depoyments and processes.
+
+### Kluctl
+
+Kluctl also uses CRs ([KluctlDeployment]({{% ref "docs/gitops/spec/v1beta1/kluctldeployment/" %}}), but only as a
+bridge between Kluctl deployment projects and GitOps. The actual project structure is solely defined via the Kluctl
+project and deployment YAML files found inside your Git repository. The KluctlDeployment CR does not add anything
+special to the deployment project itself, meaning that you never get fully dependent on in-cluster reconciliation.
+
+This means, to deploy your project, you can always revert back to using
+the [Kluctl CLI]({{% ref "docs/kluctl/commands/" %}}) even if you leverage GitOps as your main deployment strategy.
+This might sound counter intiutive at first when talking about GitOps, but there are actually very good reasons and use
+cases why you might consider mixing GitOps with other strategies. Please read the next chapter for more details on this.
+
+## Pull vs. Push
+
+GitOps can be implemented in two different strategies. One is the push strategy, which is actually what has been done
+reliably for years in classical continuous delivery, e.g. via Jenkins, Gitlab pipelines or GitHub workflows. The way it
+is usually implemented however has some disadvantages, for example the possibility of a growing drift between the state in
+Git and the state in the cluster.
+
+The other is the pull strategy, implemented via controllers running inside your target cluster. These controllers
+usually pull the Git repository in some defined interval and then (re-)apply all resources continuously. This
+reliably ensures that drift is being fixed whenever it occurs.
+
+### ArgoCD and Flux
+
+Both ways implement GitOps as a pull strategy. ArgoCD also natively supports manual synchronization, meaning that you can
+disable periodic reconciliation and instead rely on manual syncs via the UI.
```

Comment:
In Flux you can get the same behavior with `flux suspend` and `flux resume`

Reply:
True, but with some additional effort and risks involved. I'll update this part to better reflect this.

Reply:
Pushed an update of this part.